locker/memory_locker.go

```diff
@@ -0,0 +1,120 @@
+package locker
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/featureform/fferr"
+	"github.com/google/uuid"
+)
+
+type memoryKey struct {
+	id             string
+	key            string
+	ExpirationTime time.Time
+	Channel        *chan error
+}
+
+func (k *memoryKey) ID() string {
+	return k.id
+}
+
+func (k *memoryKey) Key() string {
+	return k.key
+}
+
+type MemoryLocker struct {
+	LockedItems sync.Map
+	Mutex       *sync.Mutex
+}
+
+func (m *MemoryLocker) Lock(key string) (Key, fferr.GRPCError) {
+	if key == "" {
+		return &memoryKey{}, fferr.NewInternalError(fmt.Errorf("key is empty"))
```

Comment:
Should make the message more specific or make a custom error. Going to be confusing if we just get a message saying key is empty